### PR TITLE
Contact picking

### PR DESCRIPTION
**Pull request by michaelkirk-signal**
For now this doesn't actually send a contact message, it just passes the picked contact back to the conversation view controller.

**New Contact Picker for contact shares**

picker-contact-send

**Regression check on old "invite flow"**

picker-invite

PTAL @charlesmchen

### CODE REVIEW

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m [hunk 1]
```diff
@@ -3576,6 +3590,18 @@ - (void)attachmentButtonPressed
...
     [actionSheetController addAction:gifAction];
 
+    UIAlertAction *chooseContactAction = [UIAlertAction
+        actionWithTitle:NSLocalizedString(@"ATTACHMENT_MENU_CONTACT_BUTTON", @"attachment menu option to send contact")
+                  style:UIAlertActionStyleDefault
+                handler:^(UIAlertAction *_Nonnull action) {
+                    [self chooseContactForSending];
+                }];
+    // TODO - proper image
+    UIImage *chooseContactImage = [UIImage imageNamed:@"actionsheet_camera_black"];
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: I added this to the design qs doc.

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m [hunk 2]
```diff
@@ -4848,6 +4873,37 @@ - (void)collectionView:(UICollectionView *)collectionView
...
+{
+    DDLogDebug(@"%@ in %s with contact: %@", self.logTag, __PRETTY_FUNCTION__, contact);
+
+    // TODO actually build contact message.
+    self.inputToolbar.messageText = contact.fullName;
+}
+
+- (void)contactsPicker:(ContactsPicker *)contactsPicker didSelectMultipleContacts:(NSArray<Contact *> *)contacts
+{
+    DDLogDebug(@"%@ in %s with contacts: %@", self.logTag, __PRETTY_FUNCTION__, contacts);
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: Shouldn't this be `OWSFail()`?  And shouldn't we configure the contact picker to be "single only" for contact shares?

File: Signal/src/views/ContactCell.swift [hunk 3]
```diff
@@ -39,53 +63,55 @@ class ContactCell: UITableViewCell {
     }
```
Comment (charlesmchen-signal) [inline Signal/src/views/ContactCell.swift]: Can you please attach screenshots with dynamic text size on minimum, default and maximum?

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m (hunk 2 above)
Comment (michaelkirk-signal) [reply Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: > Shouldn't this be OWSFail()

Yup, added.

> And shouldn't we configure the contact picker to be "single only" for contact shares?

Yup, it is configured via the initializer with the `multiSelection:NO` parameter.

**Pull request closed without merging**